`utils.symbols.export_all`导出的def文件包含了不必要的入口

> Note: If you use the fuzzy expressions such as 'can't use/no effect/problem/error', but don't give any relevant auxiliary information according to the template below, this issue will be not replied.

### Describe the bug
`utils.symbols.export_all`导出的def文件包含了不必要的入口

### Expected behavior
通过`utils.symbols.export_all`及如下xmake.lua能正常编译marisa-trie。

https://github.com/s-yata/marisa-trie/releases/tag/v0.2.6
```
set_project("marisa")

add_rules("mode.debug", "mode.release")

set_warnings("all", "all")

if not is_plat("windows") then
   add_cxflags("-Weffc++ -Wextra -Wconversion")
end

add_includedirs("$(projectdir)/include")
add_includedirs("$(projectdir)/lib")

target("io")
    set_kind("static")
    add_files("lib/marisa/grimoire/io/*.cc")

target("trie")
    set_kind("static")
    add_files("lib/marisa/grimoire/trie/*.cc")

target("vector")
    set_kind("static")
    add_files("lib/marisa/grimoire/vector/*.cc")

target("marisa")
    set_kind("shared")
    add_rules("utils.symbols.export_all", {export_classes = true})
    add_files("lib/marisa/*.cc")
    --add_files("lib/marisa/grimoire/io/*.cc")
    --add_files("lib/marisa/grimoire/trie/*.cc")
    --add_files("lib/marisa/grimoire/vector/*.cc")
    add_deps("io", "trie", "vector")
    on_load(function (target)
        print(target:objectfiles())
    end)
    before_install(function (package)
        os.cp("include/**.h", path.join(package:installdir()), {rootdir = "."})
    end)

target("cmdopt")
    set_kind("object")
    add_files("tools/cmdopt.cc")

target("marisa-benchmark")
    set_kind("binary")
    add_files("tools/marisa-benchmark.cc")
    add_deps("marisa", "cmdopt")
```

### Error output
If applicable, please add verbose the errors output to help us analyze your problem.
```
build\.gens\marisa\windows\x64\release\rules\symbols\export_all.def : warning LNK4102: 导出删除的析构函数“public: virtual void * __ptr64 __cdecl std::exception::`vector deleting destructor'(unsigned int) __ptr64”；映像可能不能正确运行

...

marisa.exp : error LNK2001: 无法解析的外部符号 "public: virtual void * __cdecl std::exception::`vector deleting destructor'(unsigned int)" (??_Eexception@std@@UEAAPEAXI@Z)
```


> If it is a xmake related compilation problem, please add the `-v -D` parameter and give the detailed compilation output information.

### Related Environment

Please provide compiling and running environment information:



- xmake version: 2.5.5
- os: windows
- target platform: windows

### Additional context

cmake的export功能里，是删除了`??_G`和`??_E`开头的入口，以及常量入口。

--- a/xmake/rules/utils/symbols/export_all/export_all.lua
+++ b/xmake/rules/utils/symbols/export_all/export_all.lua
@@ -66,7 +66,13 @@ function main (target, opt)
                                     symbol = symbol:sub(2)
                                 end
                                 if export_classes or not symbol:startswith("?") then
-                                    allsymbols:insert(symbol)
+                                    if export_classes then
+                                        if not symbol:startswith("??_G") and not symbol:startswith("??_E") then
+                                            allsymbols:insert(symbol)
+                                        end
+                                    else
+                                        allsymbols:insert(symbol)
+                                    end
                                 end
                             end
                         end

